### ISSUE TITLE
Segmentation fault with v0.2.28.2

### ISSUE THREAD

**Issue by timtn** [philippe44/AirConnect#290]
Hi Philippe,

I updated my airupnp from v0.2.27.0 to v0.2.28.2 this weekend, but keep getting a segmentation fault when trying to airplay to my airupnp speaker:
```
$ ./airupnp-x86-64 -c mp3
[15:38:05.662997] main:1390 Starting airupnp version: v0.2.28.2 (Oct 24 2020 @ 13:37:51)
[15:38:05.663043] main:1398 no config file, using defaults
[15:38:05.665758] Start:1108 Binding to 192.168.10.217:49153
[15:38:09.048898] AddMRDevice:992 [0x7fac9af1d010]: adding renderer (Pioneer)
[15:38:09.048990] AddMRDevice:1013 [0x7fac9af1d010]: cannot get mac Pioneer+, creating fake 3b9cb21c
[15:38:09.061746] MasterHandler:656 [0x7fac9af1d010]: subscribe success
[15:38:09.072599] XMLGetFirstDocumentItem:1370 ixmlNode_getNodeValue returned NULL
[15:38:12.635496] rtsp_thread:351 got RTSP connection 14
[15:38:14.514608] handle_rtsp:399 [0x7fac84001a90]: challenge g2fhntXKWJuBR2uFkLU+Sg==
[15:38:14.534577] handle_rtsp:392 [0x7fac84001a90]: received ANNOUNCE
[15:38:14.536258] handle_rtsp:591 [0x7fac84001a90]: responding:
RTSP/1.0 200 OK
Audio-Jack-Status: connected; type=analog
CSeq: 4


[15:38:14.544347] handle_rtsp:392 [0x7fac84001a90]: received SETUP
[15:38:14.547532] handle_rtsp:591 [0x7fac84001a90]: responding:
RTSP/1.0 200 OK
Transport: RTP/AVP/UDP;unicast;mode=record;control_port=49455;timing_port=43812;server_port=45316
Session: DEADBEEF
Audio-Jack-Status: connected; type=analog
CSeq: 5


[15:38:14.552471] handle_rtsp:392 [0x7fac84001a90]: received RECORD
Segmentation fault (core dumped)
```

Same result with the osx-multi and arm5 builds from Apple Music on iOS 14.0.1

Interestingly, Music on MacOS will get a little further and crash right after receiving the metadata:
```
handle_rtsp:392 [0x7f9680001af0]: received SET_PARAMETER
Segmentation fault (core dumped)
```
Rolling back to v0.2.28.1 seems to get rid of the the segmentation fault for me.

**Comment by philippe44** [comment]
ah crap sorry - I'll fix that tomorrow first time. It's a bit late for me today and I'm tried.

**Comment by philippe44** [comment]
I was trying to rationalize usage of strcasestr and forgot a few NULL cases . Should be corrected in 0.2.28.3

**Comment by timtn** [comment]
Thanks @philippe44 
0.2.28.3 is working perfectly!
